### ISSUE TITLE
Alignment action delayed in command chain

### ISSUE THREAD

**Issue by noctuid** [phillipberndt/pqiv#145]
When updating to the latest version of pqiv, the cycle keybindings I mentioned in https://github.com/phillipberndt/pqiv/issues/128#issuecomment-388651727 broke for me. They work for the first few times and then the alignment command is delayed after going to the next file. For example, `i` will go to the next file, but it won't be aligned at the top. Only after pressing another key (any key) does the alignment action happen. Sometimes afterwards all actions will be delayed until the next keypress. This happens consistently for me but not immediately (sometimes `i` will work correctly the first 3-10 times or so).

**Comment by phillipberndt** [comment]
Dang. So there's still a race condition. Unfortunately I completely fail to reproduce this right now :/ We knew that it worked in c279e44 - it's only been 13 commits since, and none of them touches this code. Chances are your configuration changed.

> Sometimes afterwards all actions will be delayed until the next keypress.

Can you elaborate on this?


In any case, I think I'll have to come up with a more reliable mechanism for deciding when one action completed and the next can be allowed to start..

**Comment by noctuid** [comment]
I can try to git bisect it. I'll at least see if I can get everything working on an older commit. I switched from pqiv-git to pqiv since pqiv-git is not in the aur anymore. I don't know if the recipe is at all relevant.

> Can you elaborate on this?

If I press a key to scroll up, then a key to scroll down, then any other key, it will scroll up when I hit the key to scroll down and scroll down when I hit the third key.

**Comment by noctuid** [comment]
Yeah, so this is not a new issue. I've only been able to replicate this when I start X with my graphics card using [nvidia-xrun](https://github.com/Witko/nvidia-xrun). On older commits (e.g. c279e44), pqiv will actually crash (`(pqiv:7111): Gdk-ERROR **: 18:21:24.731: The program 'pqiv' received an X Window System error.`), but that may just be an unrelated issue.

**Comment by noctuid** [comment]
I sometimes get the one keypress delay issue without my cycling commands. Do you have any advice for how I could debug this?
